examples/tv-app/linux/include/channel/ChannelManager.cpp

```diff
@@ -163,16 +168,26 @@ bool ChannelManager::HandleChangeChannelByNumber(const uint16_t & majorNumber, c
                 mCurrentChannel      = channel;
             }
         }
+        index++;
     }
     return channelChanged;
 }
 
-bool ChannelManager::HandleSkipChannel(const uint16_t & count)
+bool ChannelManager::HandleSkipChannel(const int16_t & count)
 {
-    // TODO: Insert code here
-    uint16_t newChannelIndex = static_cast<uint16_t>((count + mCurrentChannelIndex) % mChannels.size());
-    mCurrentChannelIndex     = newChannelIndex;
-    mCurrentChannel          = mChannels[mCurrentChannelIndex];
+    int32_t newChannelIndex = static_cast<int32_t>(count) + static_cast<int32_t>(mCurrentChannelIndex);
+    uint16_t channelsSize   = static_cast<uint16_t>(mChannels.size());
+
+    // handle larger than maximum -> channel size case
```

Comment:
This also handles "smaller than minimum", no?
```suggestion
    // handle newChannelIndex out of range.
```

Reply:
Added.